packages/core/src/model-hooks.ts

```diff
@@ -90,6 +91,22 @@ export interface ModelHooks<M extends Model = Model, TAttributes = any> {
   beforeAssociate(data: BeforeAssociateEventData, options: AssociationOptions<any>): AsyncHookReturn;
   afterAssociate(data: AfterAssociateEventData, options: AssociationOptions<any>): AsyncHookReturn;
 
+  /**
+   * A hook that is run at the start of {@link Model.increment} and {@link Model.decrement}
+   */
+  beforeIncrementDecrement(
+    fields: { [key in keyof TAttributes]?: number },
```

Comment:
we try to avoid naming things fields, we use attributes when it represents the JavaScript name, and columns when it represents the SQL name (with a lot of legacy slowly being migrated)

Options has a "by" parameter that is redundant. This signature should be cleaned up to only have one "options" parameter, where "by" is the map of attribute -> amount